### PR TITLE
Add a Verizon parser

### PR DESCRIPTION
**Pull request by pke11y**
This PR adds an email parser for Verizon circuit maintenance notifications.

Unit/Provider tests included:
- Confirmed notification with 1 circuit
- Confirmed notifications with 4 circuits
- Cancelled notification with 2 circuits

### CODE REVIEW

File: circuit_maintenance_parser/__init__.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
     PacketFabric,
     Telia,
     Telstra,
+    Verizon,
```
Comment (chadell) [inline circuit_maintenance_parser/__init__.py:18]: could you also update Readme/Changelog?

File: circuit_maintenance_parser/parsers/verizon.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+
+class HtmlParserVerizon1(Html):
+    """Notifications Parser for Verizon notifications."""
+
+    def parse_html(self, soup, data_base):
+        """Execute parsing."""
+        data = data_base.copy()
+        try:
+            logger.debug("PARSING")
```
Comment (chadell) [inline circuit_maintenance_parser/parsers/verizon.py:23]: parsing what? maybe a more detailed message would help :)

File: circuit_maintenance_parser/providers.py [hunk 3]
```diff
@@ -152,6 +153,13 @@ class Telstra(GenericProvider):
     _default_organizer = "gpen@team.telstra.com"
 
 
+class Verizon(GenericProvider):
+    """Verizon provider custom class."""
+
+    _parser_classes: Iterable[Type[Parser]] = [HtmlParserVerizon1]
+    _default_organizer = "NO-REPLY-sched-maint@EMEA.verizonbusiness.com"
```
Comment (chadell) [inline circuit_maintenance_parser/providers.py:160]: maybe not the most "generic" email, but until we don't have other, it should work

**Review by chadell — APPROVED** [review]
It looks good! just few simple points to complete it

File: circuit_maintenance_parser/providers.py (hunk 3 above)
Comment (pke11y) [reply circuit_maintenance_parser/providers.py:160]: agreed, but the only one we have right now.

File: circuit_maintenance_parser/parsers/verizon.py (hunk 2 above)
Comment (pke11y) [reply circuit_maintenance_parser/parsers/verizon.py]: That was only temporary, but will keep and add more detail if that works. Can remove if required.

**Review by chadell — APPROVED** [review]
LGTM

File: circuit_maintenance_parser/parsers/verizon.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    @staticmethod
+    def parse_p(p_tags: ResultSet, data: Dict):
+        """Parse <p> tag."""
+        for p_tag in p_tags:
+            p_text = p_tag.text.strip()
+            if p_text.startswith("Dear"):
+                match = re.search(r"Dear (.*),", p_text)
+                if match:
+                    data["account"] = match.group(1)
+                    break
```
Comment (glennmatthews) [inline circuit_maintenance_parser/parsers/verizon.py:81]: Perhaps simpler?

```suggestion
            p_text = p_tag.text.strip()
            match = re.match(r"Dear (.*),", p_text)
            if match:
                data["account"] = match.group(1)
                break
```

**Review by glennmatthews — APPROVED** [review]

**Review by chadell — APPROVED** [review]
LGTM

**Pull request merged**